recipe/meta.yaml

```diff
@@ -1,34 +1,79 @@
+# Notes: this receipt is based on torchvision-feedstock: https://github.com/conda-forge/torchvision-feedstock/blob/master/recipe/meta.yaml
+
 {% set name = "torchani" %}
-{% set version = "2.2" %}
+{% set version = "2.2.1" %}
+{% set torchani_proc_type = "cuda" if cuda_compiler_version != "None" else "cpu" %}
 
 package:
   name: {{ name|lower }}
   version: {{ version }}
 
 source:
-  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/{{ name }}-{{ version }}.tar.gz
-  sha256: c086fe6cabe49a0408d1f1b12aaf14e33b9e6ef0ffb15f3020792934e7c9fbbe
-  
+  git_url: https://github.com/aiqm/{{ name }}.git
+  git_rev: {{ version }}
+
 build:
-  noarch: python
   number: 0
-  script: "{{ PYTHON }} -m pip install . -vv"
+  skip: true  # [win]
+  string: py{{ CONDA_PY }}h{{ PKG_HASH }}_{{ PKG_BUILDNUM }}_{{ torchani_proc_type }}                                                  # [cuda_compiler_version == "None"]
+  string: py{{ CONDA_PY }}cuda{{ cuda_compiler_version|replace('.', '') }}h{{ PKG_HASH }}_{{ PKG_BUILDNUM }}_{{ torchani_proc_type }}  # [cuda_compiler_version != "None"]
+  script:
+    # cuda 10.2 does not have cub, which need to be manually downloaded
+    # - sudo yum install wget unzip -y                                                  # [cuda_compiler_version == '10.2']
+    # However as the current master version of cub has conflict with cuda 10.2, cuaev will not support cuda 10.2 for now.
+    - "{{ PYTHON }} -m pip install . -vvv --no-deps"                                    # [cuda_compiler_version in ("None", '10.2')]
+    - "{{ PYTHON }} -m pip install . --global-option='--cuaev-all-sms' -vvv --no-deps"  # [cuda_compiler_version not in ("None", '10.2')]
 
 requirements:
+  build:
+    - {{ compiler('c') }}
+    - {{ compiler('cxx') }}
+    - {{ compiler('cuda') }}                 # [cuda_compiler_version != "None"]
+    - sysroot_linux-64 ==2.17                # [linux64]
+
   host:
     - python
     - pip
     - setuptools_scm
+    - importlib_metadata
+    - numpy
+    - pytorch
+    - pytorch-gpu {{ pytorch_version }}  # [cuda_compiler_version != "None"]
+    - pytorch-cpu {{ pytorch_version }}  # [cuda_compiler_version == "None"]
+
   run:
     - python
-    - setuptools  # required for pkg_resources
-    - pytorch
+    - pip
+    - importlib_metadata
     - lark-parser
     - requests
+    - h5py
+    - numpy
+    - torchvision >=0.10
+    - pytorch
+    - pytorch-gpu {{ pytorch_version }}  # [cuda_compiler_version != "None"]
+    - pytorch-cpu {{ pytorch_version }}  # [cuda_compiler_version == "None"]
 
 test:
   imports:
     - torchani
+    - torchani.cuaev                     # [cuda_compiler_version != "None"]
+  source_files:
+    - torchani/resources/*
+    - tests/*
+    - tools/*
+    - download.sh
+    - dataset/*
+    - test_requirements.txt
+  commands:
+    - python -m pip install -r test_requirements.txt                          # [linux64]
```

Comment:
Please use

```
test:
  requires:
    - my_package_for_testing_1
    - my_package_for_testing_2
```

I don't think conda forge allows using pip packages but since this is for testing only it might actually be ok.

Need feedback here.

Reply:
This is for testing only, using `test_requirements.py` is easier to maintain, so that we don't need to manually update the testing required packages here.

Reply:
The number of deps is not that important and I think they are all on conda forge:

```
tb-nightly
tqdm
ase
h5py
pillow
pkbar
pyyaml
pytest
expecttest
```

Would be best to use conda for this as tests should reflect running a package using other conda packages as deps (not pip packages).

Edit: `pkbar` seems to be missing. We could make a conda package from it or just use `pip install` for this particular one.

Reply:
> so that we don't need to manually update the testing required packages here.

You'll need to update the other non-tests deps manually anyway.

Reply:
Fixed!

Reply:
`tb-nightly` should probably be `tensorboard`. If nightly is required then install it with `pip`.